### PR TITLE
Onboarding: Country selector search

### PR DESCRIPTION
**Pull request by hafizrahman**
Part of: https://github.com/woocommerce/woocommerce-android/issues/9702

### Description

This PR adds search functionality on the country list picker. See demo video below:

[country-picker-search-demo.webm](https://github.com/woocommerce/woocommerce-android/assets/266376/1893341c-8aa0-411a-80e5-859bb33bae28)

### Testing instructions

1. Start store creation,
2. Go through onboarding until the "Where is your business located?" screen is shown, 
3. Tap the current country, ensure the country picker selector is shown
4. Ensure search bar at the top is shown,
5. Try searching for various countries and ensure things work well,
6. Pick a country from the search result and ensure you're taken back to the "Where is your business located?" screen.

### CODE REVIEW

**Review by rossanafmenezes — COMMENTED** [review]
Hi @hafizrahman 

Looks good and it works as expected for the changes you made. However, the app crashes in LandScape mode when you perform a search. 

I believe we need to apply the same  selected country logic you added to the Portrait mode

```
val selectedCountry = countries.firstOrNull { it.isSelected }
            if (selectedCountry != null) {
                CountryListPickerHeader(selectedCountry)
            }
```
Otherwise, it returns a 

```
FATAL EXCEPTION: main
                 Process: com.woocommerce.android.dev, PID: 18909
                 java.util.NoSuchElementException: Collection contains no element matching the predicate.
```

**Comment by hafizrahman** [comment]
@rossanafmenezes Thank you for finding that bug! I've added a fix, and pinged @JorgeMucientes to help review further since you're in a meetup atm.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/countrypicker/CountryListPickerScreen.kt [hunk 1]
```diff
@@ -57,14 +60,36 @@ fun CountryListPickerScreen(viewModel: CountryListPickerViewModel) {
...
+        // - Scroll along with the country list on landscape mode (due to space restriction)
+
         val configuration = LocalConfiguration.current
+        val selectedCountry = countries.firstOrNull { it.isSelected }
+
         if (configuration.orientation == Configuration.ORIENTATION_PORTRAIT) {
-            CountryListPickerHeader(countries.first { it.isSelected })
+            if (selectedCountry != null) {
+                CountryListPickerHeader(selectedCountry)
+            }
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/countrypicker/CountryListPickerScreen.kt]: UI np, I think the UI of the search box is a bit iOS-ish. You can replicate other places were we added an outlined searcbox without grey background, like in `DomainPickerScreen` or `StoreProfilerScreen`, etc: 

```kotlin
    WCSearchField(
        value = searchQuery,
        onValueChange = onSearchQueryChanged,
        hint = stringResource(id = string.store_creation_store_profiler_industries_search_hint),
        modifier = modifier
            .fillMaxWidth()
            .height(52.dp)
            .border(
                BorderStroke(
                    width = dimensionResource(id = R.dimen.minor_10),
                    color = colorResource(id = R.color.woo_gray_5)
                ),
                RoundedCornerShape(dimensionResource(id = R.dimen.minor_100))
            ),
        backgroundColor = TextFieldDefaults.outlinedTextFieldColors().backgroundColor(enabled = true).value,
        keyboardOptions = KeyboardOptions(imeAction = ImeAction.Done),
        keyboardActions = KeyboardActions(onDone = { keyboardController?.hide() })
    )
```

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Hey @hafizrahman nice job! It works as expected and the landscape bug seems to be fixed 👍🏼 

I left a non blocking comment though. Should be fast to address but feel free to ignore it.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/countrypicker/CountryListPickerScreen.kt (hunk 1 above)
Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/countrypicker/CountryListPickerScreen.kt]: Very useful! Added on https://github.com/woocommerce/woocommerce-android/pull/9724/commits/f94cbda18406008753b077903d7239c66a2dde20

**Pull request merged**